flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -1070,9 +1088,9 @@ public void suspend(Throwable suspensionCause) {
 	 * exceptions that indicate a bug or an unexpected call race), and where a full restart is the
 	 * safe way to get consistency back.
 	 * 
-	 * @param t The exception that caused the failure.
+	 * @param errorInfo ErrorInfo containing the exception that caused the failure.
 	 */
-	public void failGlobal(Throwable t) {
+	public void failGlobal(ErrorInfo errorInfo) {
```

Comment:
Why don't we do a null check for `errorInfo` similar to the one in `suspend`? I think it should be fine to assume that this parameter is not null.